File: ffi/tests/dart/mock_server.dart
import 'dart:io';

import '../../../bindings/dart/lib.g.dart';

void main() {
  try {
    final server = MockServer.start();
    final uri = server.uri();
    if (!uri.startsWith('http://127.0.0.1:')) {
      stderr.writeln('unexpected uri: $uri');
      server.stop();
      exit(1);
    }
    server.stop();
  } catch (e) {
    stderr.writeln('MockServer test failed: $e');
    exit(1);
  }
}

File: ffi/tests/python/mock_server.py
import sys

from github_mock_api import MockServer


def main() -> None:
    try:
        server = MockServer.start()
        uri = server.uri()
        if not uri.startswith("http://127.0.0.1:"):
            print(f"unexpected uri: {uri}", file=sys.stderr)
            server.stop()
            sys.exit(1)

        server.stop()
    except Exception as e:
        print(f"MockServer test failed: {e}", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()
